docs/style-guide.adoc

```diff
@@ -0,0 +1,455 @@
+= The Cypher Style Guide
+
+Style guide for writing Cypher code.
+
+NOTE: Cypher is a language.
+Languages have their own rules, and are not eager to adapt to circumstance.
+Remember not to be too eager to shiv a Cypher expression into the conventions of a different language.
+
+== Why Is Style Important?
+
+Consider this dadaist work of art from Nigel Small and Nicole White:
+
+.Insane query
+[source, cypher]
+----
+MATCH (null)-[:merge]->(true)
+with null.delete as foreach, `true`.false as null
+return 2 + foreach, coalesce(null, 3.1415)
+limit 10;
+----
+
+Then compare it to this classical piece by Mark Needham:
+
+.Sane query
+[source, cypher]
+----
+MATCH (member:Member {name: 'Mark Needham'})
+      -[:HAS_MEMBERSHIP]->()-[:OF_GROUP]->(:Group)-[:HAS_TOPIC]->(topic)
+WITH member, topic, count(*) AS score
+MATCH (topic)<-[:HAS_TOPIC]-(otherGroup:Group)
+WHERE NOT (member)-[:HAS_MEMBERSHIP]->(:Membership)-[:OF_GROUP]->(otherGroup)
+RETURN otherGroup.name, collect(topic.name), sum(score) AS score
+ORDER BY score DESC
+----
+
+== Rules
+
+Whenever two rules are in conflict, use the order to decide which one trumps.
+
+// Template:
+//. rule
+//+
+//.Bad
+//[source, cypher]
+//----
+//MATCH (person:Person {property: value})
+//----
+//.. No padding space for parameters.
+//+
+//.Good
+//[source, cypher]
+//----
+//WITH { param } AS value
+//----
+
+. Start a new clause on a new line.
++
+.Bad
+[source, cypher]
+----
+MATCH (n) WHERE n.name CONTAINS 's' RETURN n.name
+----
++
+.Good
+[source, cypher]
+----
+MATCH (n)
+WHERE n.name CONTAINS 's'
+RETURN n.name
+----
+
+.. Indent subclauses `ON CREATE` and `ON MATCH` with two spaces.
++
+.Bad
+[source, cypher]
+----
+MERGE (n) ON CREATE SET n.prop = 0
+MERGE (a:A)-[:T]-(b:B)
+ON CREATE SET a.name = 'me'
+ON MATCH SET b.name = 'you'
+----
++
+.Good
+[source, cypher]
+----
+MERGE (n)
+  ON CREATE SET n.prop = 0
+MERGE (a:A)-[:T]-(b:B)
+  ON CREATE SET a.name = 'me'
+  ON MATCH SET b.name = 'you'
+----
+
+.. Put `ON CREATE` before `ON MATCH` if both are present.
+// no example necessary
+
+. Write keywords in all caps.
```

Comment:
I would suggest "write keywords in all caps." -> "Write keywords in upper case."